test/tests/functional/pbs_qstat_formats.py

```diff
@@ -621,3 +621,60 @@ def test_qstat_json_valid_job_longint_env(self):
             json.loads(qstat_out)
         except ValueError:
             self.assertTrue(False)
+
+    def run_namelength_test(self, options=''):
+        """
+        Changes the server name, sets a long job and queue name,
+        and ensures they're truncated correctly.
+        """
+        self.server.stop()
+        self.assertFalse(self.server.isUp(), 'Failed to stop PBS')
```

Comment:
This will take some time to do a negative test.  isUp() will wait I think 20s until it gives up.  I'm not sure there's a way to get around this though.

Reply:
There are other tests that do the same, I'm not sure of a better way.